### PR TITLE
Replace array constructor with literal when it's possible

### LINKED ISSUE #246: Enhancement: `new Array(5)` can be mangled into `[,,,,,]`
As indicated by the title, an `Array` constructor with a non-variable initial length can be more concisely written as a literal with blank elements:

~~~js
new Array(0)
[]
new Array(1)
[,]
new Array(2)
[,,]
new Array(3)
[,,,]

…

new Array(9)
[,,,,,,,,,]
new Array(10)
[,,,,,,,,,,]
~~~

The limit for this optimisation is 11, where both forms are of equal length:

~~~js
new Array(11)
[,,,,,,,,,,,]
~~~

As of v3.14.1, Terser isn't doing this, but it probably should.

### PR DESCRIPTION
**Pull request by L2jLiga**
Closes #246

### CODE REVIEW

**Comment by fabiosantoscode** [comment]
Nice fix!

But we can't run it unless `unsafe` is turned on. `unsafe`, which is mostly safe, is reserved for these kinds of things. The `Array` constructor might have been redeclared as a global variable and we have no way to know!

**Comment by L2jLiga** [comment]
Yes, you're right, it can be redeclared by code like `class Array { /* ... */ }`, will move it to `unsafe`

**Comment by L2jLiga** [comment]
@fabiosantoscode I updated this PR

File: lib/compress.js [hunk 1]
```diff
@@ -5303,19 +5303,25 @@ merge(Compressor.prototype, {
...
+
+        if (compressor.option("unsafe") && is_undeclared_ref(exp)) {
+            if (["Object", "RegExp", "Function", "Error"].includes(exp.name))
+                return make_node(AST_Call, self, self).transform(compressor);
+
+            if (exp.name === "Array" && args.length < 2) {
+                if (args.length === 0 || args[0].value === 0) return new AST_Array;
+
+                if (args[0].value > 11)
                     return make_node(AST_Call, self, self).transform(compressor);
```
[the remark this answers is not in the document]
Comment (fabiosantoscode) [reply lib/compress.js]: Nevermind

**Comment by fabiosantoscode** [comment]
Awesome stuff. Thanks for the effort!

**Pull request merged**